### PR TITLE
refact: added checks to Phase.addComponent

### PR DESCRIPTION
**Pull request by asmfstatoil**

### CODE REVIEW

**Comment by asmfstatoil** [comment]
image

Looks to be some mixups in terms of input order of moles and molesinphase

File: src/main/java/neqsim/thermo/phase/PhaseWax.java [hunk 1]
```diff
@@ -46,10 +46,9 @@ public void init(double totalNumberOfMoles, int numberOfComponents, int type, in
 
   /** {@inheritDoc} */
   @Override
-  public void addcomponent(String componentName, double molesInPhase, double moles,
-      int compNumber) {
-    super.addcomponent(molesInPhase);
-    componentArray[compNumber] = new ComponentWax(componentName, moles, molesInPhase, compNumber);
+  public void addcomponent(String name, double moles, double molesInPhase, int compNumber) {
+    super.addcomponent(name, moles);
```
Comment (asmfstatoil) [inline src/main/java/neqsim/thermo/phase/PhaseWax.java:50]: should be addcomponent(name, molesInPhase?)

**Comment by asmfstatoil** [comment]
@EvenSol can you review my two review comments?

File: src/main/java/neqsim/thermo/phase/PhaseWax.java (hunk 1 above)
Comment (EvenSol) [reply src/main/java/neqsim/thermo/phase/PhaseWax.java:50]: I did a check. You are right. It should be addcomponent(name, molesInPhase)

File: src/main/java/neqsim/thermo/phase/PhaseSolid.java [hunk 2]
```diff
@@ -51,10 +51,9 @@ public void init(double totalNumberOfMoles, int numberOfComponents, int type, in
 
   /** {@inheritDoc} */
   @Override
-  public void addcomponent(String componentName, double molesInPhase, double moles,
-      int compNumber) {
-    super.addcomponent(molesInPhase);
-    componentArray[compNumber] = new ComponentSolid(componentName, moles, molesInPhase, compNumber);
+  public void addcomponent(String name, double moles, double molesInPhase, int compNumber) {
+    super.addcomponent(name, moles);
```
[the remark this answers is not in the document]
Comment (EvenSol) [reply src/main/java/neqsim/thermo/phase/PhaseSolid.java:55]: I did a check. You are right. It should be addcomponent(name, molesInPhase)

File: src/main/java/neqsim/thermo/phase/PhaseEos.java [hunk 3]
```diff
@@ -214,41 +208,31 @@ public double molarVolume2(double pressure, double temperature, double A, double
...
 
     setMolarVolume(1.0 / BonV * Btemp / numberOfMolesInPhase);
     int iterations = 0;
     int maxIterations = 1000;
-
     do {
       iterations++;
       BonVold = BonV;
-      h = BonV + Btemp * gV() + Btemp * Dtemp / (numberOfMolesInPhase * temperature) * fv()
+      double h = BonV + Btemp * gV() + Btemp * Dtemp / (numberOfMolesInPhase * temperature) * fv()
```
Comment (EvenSol) [inline src/main/java/neqsim/thermo/phase/PhaseEos.java:220]: will this increase calculation time (making doubles inside the while loop)?

Comment (EvenSol) [reply src/main/java/neqsim/thermo/phase/PhaseEos.java:220]: I guess the only reason to have it as before is speed (this loop is run a lot of times in all types of calculations) - I guess in any calculations a significant amount of time is spent on this method.

Comment (asmfstatoil) [reply src/main/java/neqsim/thermo/phase/PhaseEos.java:220]: I was hoping the compiler would optimize the code such that it did not matter. 

https://stackoverflow.com/questions/8803674/declaring-variables-inside-or-outside-of-a-loop

**Pull request merged**